src/json-patch-duplex.ts

```diff
@@ -148,30 +191,26 @@ module jsonpatch {
       obj[key] = this.value;
       return removed;
     },
-    move: function (obj, key, tree) {
-      var getOriginalDestination: any = { op: "_get", path: this.path };
-      apply(tree, [getOriginalDestination]);
-      // In case value is moved up and overwrites its ancestor
-      var original = getOriginalDestination.value === undefined ?
-        undefined : JSON.parse(JSON.stringify(getOriginalDestination.value));
+    move: function (obj, key, document) {
+      const originalValue = getValueByPointer(document, this.path);
```

Comment:
Can't you use `result` of `const originalValue = applyOperation(document, { op: "remove", path: this.from } ).result;` so you don't have to apply `_get` here and iterate over tree once again?